pkg/controller/spoketoken/spoke_token_controller.go

```diff
@@ -244,27 +254,27 @@ func (r *ReconcileAgentToken) prepareAgentTokenSecret(token string) *corev1.Secr
 	return mcSecret
 }
 
-func (r *ReconcileAgentToken) getServiceAccountTokenSecret(sa corev1.ServiceAccount) *corev1.Secret {
-	// Get the service account token from the service account's secret list
-	saSecret := &corev1.Secret{}
-
-	for _, secret := range sa.Secrets {
-		secretName := types.NamespacedName{
-			Name:      secret.Name,
-			Namespace: sa.Namespace,
-		}
+func (r *ReconcileAgentToken) getServiceAccountTokenSecret() *corev1.Secret {
+	// Get all secrets
+	// list thing for rolling update check
+	secretList := &corev1.SecretList{}
+	listopts := &client.ListOptions{Namespace: "open-cluster-management-agent-addon"}
+	err := r.Client.List(context.TODO(), secretList, listopts)
 
-		if err := r.Client.Get(context.TODO(), secretName, saSecret); err != nil {
-			continue
-		}
+	if err != nil {
+		klog.Error(err.Error())
+		return nil
+	}
 
-		// Get the service account token
-		if saSecret.Type == corev1.SecretTypeServiceAccountToken {
-			break
+	for _, secret := range secretList.Items {
+		// Get the application-manager service account token
+		if secret.Type == corev1.SecretTypeServiceAccountToken && strings.HasPrefix(secret.Name, "application-manager-token-") {
```

Comment:
Since we are no longer relying on a direct linkage of ServiceAccount to Secret and instead just relying on the type and the name of the secret. It seems like if there is a secret with a prefix  `application-manager-token-` and the right type that anyone can create it will run into some kind of conflict?

@rokej I assume this issue is not unique to the app team. Is there another PR from another team that we can reference? Thanks.

Reply:
Good point, I will add another annotation check `kubernetes.io/service-account.name: application-manager`.  Here is another PR foundation team did https://github.com/stolostron/managedcluster-import-controller/pull/177

Reply:
@mikeshng I don't think you can create a service account type secret without actual service account.

Reply:
That will be a kubernetes bug.